libraries/WiFi/src/WiFiClass.cpp

```diff
@@ -88,40 +135,27 @@ int WiFiClass::beginBSSID(const char* ssid, const uint8_t *bssid) {
           must be between ASCII 32-126 (decimal).
 */
 int WiFiClass::begin(const char* ssid, const char *passphrase, const uint8_t *bssid) {
-    // Simple ESP8266 compatibility hack
-    if (_modeESP == WIFI_AP) {
-        return beginAP(ssid, passphrase);
-    }
-
-    end();
-
-    _ssid = ssid;
-    _password = passphrase;
-    if (bssid) {
-        memcpy(_bssid, bssid, sizeof(_bssid));
-    } else {
-        bzero(_bssid, sizeof(_bssid));
-    }
-    _wifi.setSTA();
-    _wifi.setSSID(_ssid.c_str());
-    _wifi.setBSSID(_bssid);
-    _wifi.setPassword(passphrase);
-    _wifi.setTimeout(_timeout);
-    _apMode = false;
-    _wifiHWInitted = true;
     uint32_t start = millis(); // The timeout starts from network init, not network link up
-    if (!_wifi.begin()) {
+
+    // Returns WL_IDLE_STATUS on error for compatibility.
+    if (_beginInternal(ssid, passphrase, bssid)) {
         return WL_IDLE_STATUS;
     }
-    noLowPowerMode();
-    // Enable CYW43 event debugging (make sure Debug Port is set)
```

Comment:
Could you please move this commented 2 lines (about CYW43 tracing) to the ::beginInternal method?  I use this very occasionally while debugging the PicoW, but not enough to remember how to do it w/o the prompt. :)